docs/micro-service-architecture.md

```diff
@@ -0,0 +1,248 @@
+community support compared to more established solutions like Keycloak or Auth0.
+- Pricing: While Zitadel offers a free tier, pricing for larger enterprises might be a consideration.
+
+#### Auth0
+**Pros**
+- Ease of Use: Auth0 is known for its ease of use and quick setup. It provides a user-friendly dashboard and supports various identity scenarios out of the box.
+- Large User Base: Auth0 is widely adopted and has a large user base, leading to a robust community and extensive third-party integrations.
+- Extensive Documentation: Auth0 provides thorough and well-organized documentation, making it easier for developers to implement and troubleshoot.
+
+**Cons**
+- Pricing: Auth0's pricing might be higher compared to some other solutions, especially for large enterprises with high user counts.
+- Limited Customization in Free Tier: Some advanced features and customization options may be restricted in the free tier.
+
+
+## Services
+![microservices](./ms.png)
+
+### `API Gateway`
+
+The API Gateway is a single entry point for all clients. 
+It provides HTTP endpoints to access the different services in the infrastructures, and does so while checking the authorization claims of the user doing the request. 
+This service works closely with the Keycloak service.
+
+
+### `Keycloak`
+
+It will allow us to provide a secure authentication and authorization service for the client. 
+
+### `Notification Service`
+
+The Notification Service is responsible for sending notifications to the users. The goal is to have a single service responsible for sending notifications to the users.
+
+### `Job Service`
+
+The Job offer service is responsible for managing jobs, job offers and companies. This service is implemented in kotlin using spring r2dbc as a reactive database driver. 
+We chose r2dbc instead of jpa since being we needed to have asynchronous requests in our architecure to optimize latency. 
+Having a synchronous system would mean we can only treat one request at the time which at larger scale will not be efficient enough. 
+The service uses a postgres database to store it's data.
+
+### `Employer Service`
+
+The Employer Service represents the external software that employers uses to manage their job offers. 
+
+LinkedOut calls to this external software for retrieving job offers, or giving evaluations to employers for instance. 
+On the other hand, this external software calls to LinkedOut when an employer wants to message a seasonal worker, or to give an evaluation to a seasonal worker for instance.
+
+Since the employer service is normally an external service, we will only implement something that mocks it.
+
+### `Profile Service`
+
+The Profile Service is responsible for managing the profiles of the users. This service will be written in Kotlin as a matter of uniformity with the other services. 
+Profiles will be stored in a Postgres database and also uses r2dbc as it's database driver.
```

Comment:
No need to say that we write it in Kotlin as we say so in the "Technologies" section.

```suggestion
The Profile Service is responsible for managing the profiles of the users.

Profiles will be stored in a Postgres database and also uses R2DBC as its database driver.
```